llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -352,9 +358,37 @@ bool isEntryPoint(const Function &F) {
   return F.hasFnAttribute("sycl-module-id") && !isSpirvSyclBuiltin(F.getName());
 }
 
+unsigned getAspectId(std::string Aspect) {
+#define __SYCL_ASPECT(ASPECT, ID)                                              \
+  if (Aspect == #ASPECT) {                                                     \
+    return ID;                                                                 \
+  }
+#include <sycl/info/aspects.def>
+#undef __SYCL_ASPECT
+  return INT_MAX;
+}
```

Comment:
We shouldn't do this. Requiring SYCL headers in a pass make it so we can't build a compiler without having headers and compiler should be independent from runtime implementation, see #5892.

In #6804 we implemented a solution, where FE emits "string" -> "id" mapping for aspects defined by the runtime. We should use that mapping here instead of relying on RT headers directly

Reply:
[3c69169](https://github.com/intel/llvm/pull/7115/commits/3c691692a7c5e159f9a06af42f32c28c3c729f90) thanks, I didn't notice that Steffen already done this mapping.